sky/cli.py

```diff
@@ -3127,10 +3147,10 @@ def _output():
                 if not show_all:
                     accelerator_table_vals.append(region_str)
                 accelerator_table.add_row(accelerator_table_vals)
-
-            if i != 0:
+            if i != 0 and num_unique_accelerators > 0:
                 yield '\n\n'
             yield from accelerator_table.get_string()
+            num_unique_accelerators -= 1
```

Comment:
Should not be required once we update `list_accelerators` to return empty dict if resources do not match pass the quantity filter.